zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/group/GroupTest.java

```diff
@@ -49,4 +49,38 @@ public void shouldNotDuplicate() {
             "Expected to create group with name '%s', but a group with this name already exists."
                 .formatted(name));
   }
+
+  @Test
+  public void shouldUpdateGroup() {
+    // given
+    final var name = UUID.randomUUID().toString();
+    final var groupRecord = engine.group().newGroup(name).create();
+
+    // when
+    final var groupKey = groupRecord.getKey();
+    final var updatedName = name + "-updated";
+    final var updatedGroupRecord =
+        engine.group().updateGroup(groupKey).withName(updatedName).update();
+
+    final var updatedGroup = updatedGroupRecord.getValue();
+    Assertions.assertThat(updatedGroup)
+        .isNotNull()
+        .hasFieldOrPropertyWithValue("name", updatedName);
+  }
+
+  @Test
+  public void shouldRejectUpdatedIfNoGroupExists() {
```

Comment:
❌ I'm missing the test where the processor reject the command because you are trying to update a group with the same name